datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/timeline/mappers/SchemaFieldBlameMapper.java

```diff
@@ -0,0 +1,188 @@
+package com.linkedin.datahub.graphql.types.timeline.mappers;
+
+import com.linkedin.common.urn.Urn;
+import com.linkedin.datahub.graphql.generated.ChangeOperationType;
+import com.linkedin.datahub.graphql.generated.GetSchemaBlameResult;
+import com.linkedin.datahub.graphql.generated.SchemaFieldBlame;
+import com.linkedin.datahub.graphql.generated.SchemaFieldChange;
+import com.linkedin.datahub.graphql.generated.SemanticVersionChangeType;
+import com.linkedin.datahub.graphql.generated.SemanticVersionStruct;
+import com.linkedin.metadata.key.SchemaFieldKey;
+import com.linkedin.metadata.timeline.data.ChangeCategory;
+import com.linkedin.metadata.timeline.data.ChangeEvent;
+import com.linkedin.metadata.timeline.data.ChangeTransaction;
+import com.linkedin.metadata.utils.EntityKeyUtils;
+import com.linkedin.util.Pair;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.parquet.SemanticVersion;
+
+
+// Class for converting ChangeTransactions received from the Timeline API to SchemaFieldBlame structs for every schema
+// at every semantic version.
+@Slf4j
+public class SchemaFieldBlameMapper {
+
+  public static GetSchemaBlameResult map(List<ChangeTransaction> changeTransactions, @Nullable String versionCutoff) {
```

Comment:
Just to confirm, changeTransactions will never be null right?

Reply:
Yes, will never be null.